File: backend/product_service/src/main/java/ucv/app_inventory/application/DTO/CategoryDTO.java
package ucv.app_inventory.application.DTO;

import java.io.Serializable;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ucv.app_inventory.domain.entities.Category.Status;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CategoryDTO implements Serializable {

    @JsonProperty("id_categoria")
    private Long id;

    @JsonProperty("nombre")
    private String name;

    @JsonProperty("descripcion")
    private String description;

    @JsonProperty("estado")
    private Status status;
}

File: frontend/src/products/infraestructure/dto/CategoryDTO.js
export class CategoryDTO {
    constructor({ id, name, description, status }) {
      this.id = id;
      this.name = name;
      this.description = description;
      this.status = status;
    }
  
    static fromDomain(category) {
        return new CategoryDTO({ ...category });
      }
    
      toDomain() {
        return { ...this };
      }
  }
